required/latex-lab/latex-lab-toc-kernel-changes.dtx

```diff
@@ -181,15 +179,12 @@
 % \cs{addcontentsline} should always write the unique counter representation
 % \cs{@currentHref} so that we can create the |/Ref| key. At the same time
 % we need a hook, so that hyperref can add its bookmarks code. The generic hook
-% would be ok, but we need the arguments, so we store them first.
+% is ok. We only declare it and add until the merge.
 %
 %    \begin{macrocode}
-\NewHook{cmd/addcontentsline/before}
-\def\addcontentsline#1#2#3{% toc extension, type, tag
-    \def\addcontentsline@arg@one{#1}% name??
-    \def\addcontentsline@arg@two{#2}%
-    \def\addcontentsline@arg@three{#3}%
-    \UseHook{cmd/addcontentsline/before}%
+\NewHookWithArguments{cmd/addcontentsline/before}{3}
+\def\addcontentsline#1#2#3{% #1=toc extension, #2= heading type, tag
+    \UseHookWithArguments{cmd/addcontentsline/before}{3}{#1}{#2}{#3}%
```

Comment:
we should review if we really want to turn individual generic hooks into hooks with arguments. I think we should not and instead use \AddTohookWithArguments{addcontentsline} here and leave the generic one for code that goes around the whole command for one or the other reason. It is not less efficient (ie when the generic hook is not filled it is not patched in)

Reply:
as I said: I think we should not  change generic hooks to become hooks with arguments. I think this leads to bad style and difficulties. Why is the generic hook filled? It is only filled by the tagging code so if that goes into a hook by itself then the generic hook would normally not be there and would be free to be used on the outside for somehting else if needed.

Reply:
I already removed the declaration and the UseHook command hook.  (But I don't quite understand the distinction, we just made generic hooks into hooks with argument).